src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -1120,20 +1122,79 @@ void SpatialPooler::updateDutyCyclesHelper_(vector<Real>& dutyCycles,
 
 void SpatialPooler::updateBoostFactors_()
 {
-  for (UInt i = 0; i < numColumns_; i++)
+  if (maxBoost_ > 1.0)
+  {    
+    if (globalInhibition_)
+    {
+      updateBoostFactorsGlobal_();
+    }
+    else
+    {
+      updateBoostFactorsLocal_();
+    }
+  }
+}
+
+void SpatialPooler::updateBoostFactorsGlobal_()
+{
+  Real targetDensity;
+  if (numActiveColumnsPerInhArea_ > 0)
+  {
+    UInt inhibitionArea = pow((Real) (2 * inhibitionRadius_ + 1),
+                              (Real) columnDimensions_.size());
+    inhibitionArea = min(inhibitionArea, numColumns_);
+    targetDensity = ((Real) numActiveColumnsPerInhArea_) / inhibitionArea;
+    targetDensity = min(targetDensity, (Real) 0.5);
+  }
+  else
+  {
+    targetDensity = localAreaDensity_;
+  }
+
+  for (UInt i = 0; i < numColumns_; ++i)
+  {
+    Real boostFactor = exp(-(activeDutyCycles_[i] - targetDensity)
+                           * maxBoost_);
+
+    // Avoid floating point mismatches between implementations.
+    boostFactors_[i] = round(boostFactor * 100.0) / 100.0;
```

Comment:
One way to avoid float precision errors is to just use integers instead. It's a little less readable since you have to read comments to understand what the scale means but avoids lines like this. I wouldn't recommend making this fairly large refactor in this PR, just pointing it out as a perhaps cleaner way to implement things.

Also, you might be able to avoid this line if you use the fixed precision variables types (`UInt32`, `Real32`).